src/specs/telemetry.spec.js

```diff
@@ -3,68 +3,83 @@ const { handlers, functions, cmds } = require('./effects')
 const { basicMultistep, badHandler, basic } = functions
 const { sleep } = require('./test-util')
 
-test('telemetry - onCommandComplete', async () => {
-  let telemetry = []
-  const onCommandComplete = t => {
-    telemetry.push(t)
-  }
-  const config = { onCommandComplete, name: 'telemetry' }
-  const now = Date.now()
-  await call(config, handlers, basicMultistep, 'foo')
-  expect(telemetry.length).toEqual(2)
-  telemetry.forEach((t, i) => {
-    const message = 'foo' + (i + 1)
-    expect(t.success).toEqual(true)
-    expect(t.command).toEqual(cmds.echo(message))
-    expect(t.latency).toBeLessThan(5)
-    expect(t.start).toBeGreaterThanOrEqual(now)
-    expect(t.end).toBeGreaterThanOrEqual(t.start)
-    expect(t.index).toEqual(0)
-    expect(t.step).toEqual(i)
-    expect(t.result).toEqual(message)
-    expect(t.config).toEqual(config)
+test('telemetry - onCommandComplete', () => {
+  return new Promise(async resolve => {
+    let telemetry = []
+    const onCommandComplete = t => {
+      telemetry.push(t)
+    }
+    const config = { onCommandComplete, name: 'telemetry' }
+    const now = Date.now()
+    await call(config, handlers, basicMultistep, 'foo')
+    setImmediate(() => {
+      expect(telemetry.length).toEqual(2)
+      telemetry.forEach((t, i) => {
+        const message = 'foo' + (i + 1)
+        expect(t.success).toEqual(true)
+        expect(t.command).toEqual(cmds.echo(message))
+        expect(t.latency).toBeLessThan(5)
```

Comment:
This was in the existing test, but should this just be tested as a `number`? Having `5` here seems like it might be a race condition?